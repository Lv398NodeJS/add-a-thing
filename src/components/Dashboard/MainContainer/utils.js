export const getTaskListAsArray = (snapValue = {}) => {
  const taskList = [];

  Object.keys(snapValue).forEach(task => (
    taskList.push({
      id: task,
      name: snapValue[task].name,
      priority: snapValue[task].priority,
      description: snapValue[task].description,
      status: snapValue[task].status,
      key: task,
    })));

  return taskList;
};
