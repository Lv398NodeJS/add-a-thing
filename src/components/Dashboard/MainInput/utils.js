export const addTaskWithHash = inputData => inputData.slice(0, inputData.length - 2).trim();

export const showPriorityColor = (newPriority) => {
  let style = '';
  switch (newPriority) {
    case 'High':
      style += 'priorityH';
      break;
    case 'Medium':
      style += 'priorityM';
      break;
    case 'Low':
      style += 'priorityL';
      break;
    default:
      break;
  }
  return style;
};

export const changePriority = (getPriority) => {
  let setPriority = '';
  if (getPriority === '1') {
    setPriority = 'High';
  } else if (getPriority === '3') {
    setPriority = 'Low';
  } else {
    setPriority = 'Medium';
  }
  return setPriority;
};
