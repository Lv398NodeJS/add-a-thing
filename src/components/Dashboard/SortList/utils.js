export const sort = (array, field, direction = 'ASC') => {
  const copiedArray = [...array]; // For immutability

  copiedArray.sort((a, b) => {
    if (direction === 'NONE') {
      return 0;
    }

    let compareA = String(a[field]).toLowerCase();
    let compareB = String(b[field]).toLowerCase();

    if (field === 'priority') {
      compareA = ['Low', 'Medium', 'High'].indexOf(compareA);
      compareB = ['Low', 'Medium', 'High'].indexOf(compareB);
    }

    return compareA.localeCompare(compareB);
  });

  if (direction === 'DESC') {
    copiedArray.reverse();
  }

  return copiedArray;
};

export const storage = {
  get: (key) => {
    let data = localStorage.getItem('SortList_Data') || '{}';
    data = JSON.parse(data);
    return data[key];
  },

  set: (key, value) => {
    let data = localStorage.getItem('SortList_Data') || '{}';
    data = JSON.parse(data);
    data[key] = value;
    data = JSON.stringify(data);
    localStorage.setItem('SortList_Data', data);
  },
};
