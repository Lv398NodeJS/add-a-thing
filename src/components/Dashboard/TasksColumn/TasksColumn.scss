.tasks-column-todo-title,
.tasks-column-in-progress-title,
.tasks-column-done-title,
.tasks-column-invalid-title {
  text-align: center;
  font-size: 1.8rem;
  user-select: none;
}

.tasks-column-todo-title {
  color: rgb(103, 102, 106);
  border-bottom: 3px solid rgb(103, 102, 106);
}

.tasks-column-in-progress-title {
  color: rgb(118, 183, 192);
  border-bottom: 3px solid rgb(116, 154, 159);
}

.tasks-column-done-title {
  color: rgb(194, 105, 95);
  border-bottom: 3px solid rgb(191, 134, 123);
}

.tasks-column-invalid-title {
  color: rgb(78, 26, 235);
  border-bottom: 3px solid rgb(78, 26, 235);
}

.task-items-container {
  flex: 0 1 auto;
  overflow-y: auto;
  padding: 0px;
  width: 100%;
  max-height: 460px;
}

::-webkit-scrollbar {
  display: none;
}
