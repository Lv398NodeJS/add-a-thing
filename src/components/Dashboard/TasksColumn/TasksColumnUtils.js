export const columnTitleClass = (title) => {
  switch (title) {
    case 'To Do':
      return 'tasks-column-todo-title';
    case 'In Progress':
      return 'tasks-column-in-progress-title';
    case 'Done':
      return 'tasks-column-done-title';
    default:
      return 'tasks-column-invalid-title';
  }
};

export const loaderColor = (title) => {
  switch (title) {
    case 'To Do':
      return '#67666a';
    case 'In Progress':
      return '#749a9f';
    case 'Done':
      return '#bf867b';
    default:
      return '#4e1aeb';
  }
};
