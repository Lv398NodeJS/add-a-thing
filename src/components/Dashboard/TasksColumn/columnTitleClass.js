export default (title) => {
  switch (title) {
    case 'To Do':
      return 'tasksColumnToDoTitle';
    case 'In Progress':
      return 'tasksColumnInProgressTitle';
    case 'Done':
      return 'tasksColumnDoneTitle';
    default:
      return 'tasksColumnInvalidTitle';
  }
};
