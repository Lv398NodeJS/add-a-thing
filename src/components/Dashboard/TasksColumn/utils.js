export const columnTitleClass = (title) => {
  switch (title) {
    case 'To Do':
      return 'tasks-column-todo-title';
    case 'In Progress':
      return 'tasks-column-in-progress-title';
    case 'Done':
      return 'tasks-column-done-title';
    default:
      return 'tasks-column-invalid-title';
  }
};
