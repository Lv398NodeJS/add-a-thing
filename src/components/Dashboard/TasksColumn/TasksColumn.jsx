import React, { Component } from 'react';
import { Container } from 'react-bootstrap';
import './TasksColumn.scss';
import TaskItem from '../TaskItem/TaskItem';
import SortList from '../SortList/SortList';

export default class TasksColumn extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  columnTitleClass = () => {
    const { title } = this.props;

    switch (title) {
      case 'To Do':
        return 'tasksColumnToDoTitle';
      case 'In Progress':
        return 'tasksColumnInProgressTitle';
      case 'Done':
        return 'tasksColumnDoneTitle';
      default:
        return 'tasksColumnInvalidTitle';
    }
  };

  sortIconColor = () => {
    const { title } = this.props;

    switch (title) {
      case 'To Do':
        return 'rgb(103, 102, 106)';
      case 'In Progress':
        return 'rgb(118, 183, 192)';
      case 'Done':
        return 'rgb(194, 105, 95)';
      default:
        return 'rgb(78, 26, 235)';
    }
  };

  render() {
    const {
      sortedTasks, taskListRef, title,
    } = this.props;

    const sortIconColor = this.sortIconColor();

    const tasksToDisplay = sortedTasks.map(
      task => (
        <TaskItem
          key={task.id}
          status={task.status}
          priority={task.priority}
          priorityForSorting={['Low', 'Medium', 'High'].indexOf(task.priority)}
          id={task.id}
          taskName={task.name}
          taskListRef={taskListRef}
        />
      ),
    );

    return (
      <div className="tasksColumn rounded mb-3">
        <h1 className={this.columnTitleClass()}>
          {title}
        </h1>
        <Container className="taskItemsContainer">
          <SortList
            storageKey={window.location.pathname + title}
            sortIconColor={sortIconColor}
            fields={[
              {
                key: 'taskName',
                text: 'Name',
              },
              {
                key: 'priorityForSorting',
                text: 'Priority',
              },
            ]}
          >
            {tasksToDisplay.reverse()}
          </SortList>
        </Container>
      </div>
    );
  }
}
