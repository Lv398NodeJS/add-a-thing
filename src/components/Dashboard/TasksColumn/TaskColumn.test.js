import React from 'react';
import { shallow, render } from 'enzyme';
import * as utils from './utils';
import TasksColumn from './TasksColumn';

const taskColumn = shallow(<TasksColumn sortedTasks={[]} />);

describe('<TaskColumn />', () => {
  it('should render and match snapshot', () => {
    const taskColumnR = render(<TasksColumn sortedTasks={[]} />);
    expect(taskColumnR).toMatchSnapshot();
  });
  it('should render title element', () => {
    const title = taskColumn.find('[data-test="columnTitle"]');
    expect(title.length).toEqual(1);
  });
  it('should render title text from props', () => {
    const titleText = 'To Do';
    const taskColumnT = shallow(<TasksColumn sortedTasks={[]} title={titleText} />);
    const title = taskColumnT.find('[data-test="columnTitle"]').text();
    expect(title).toEqual(titleText);
  });
  it('should have correct ToDo class name', () => {
    const titleText = 'To Do';
    const taskColumnT = shallow(<TasksColumn sortedTasks={[]} title={titleText} />);
    const title = taskColumnT.find('[data-test="columnTitle"]');
    const finalTitleStyle = utils.columnTitleClass(titleText);
    expect(title.hasClass(finalTitleStyle)).toBeTruthy();
  });
  it('should have correct InProgress class name', () => {
    const titleText = 'In Progress';
    const taskColumnT = shallow(<TasksColumn sortedTasks={[]} title={titleText} />);
    const title = taskColumnT.find('[data-test="columnTitle"]');
    const finalTitleStyle = utils.columnTitleClass(titleText);
    expect(title.hasClass(finalTitleStyle)).toBeTruthy();
  });
  it('should have correct Done class name', () => {
    const titleText = 'Done';
    const taskColumnT = shallow(<TasksColumn sortedTasks={[]} title={titleText} />);
    const title = taskColumnT.find('[data-test="columnTitle"]');
    const finalTitleStyle = utils.columnTitleClass(titleText);
    expect(title.hasClass(finalTitleStyle)).toBeTruthy();
  });
  it('should render container for tasks', () => {
    const tasksCont = taskColumn.find('[data-test="taskItemsContainer"]');
    expect(tasksCont.length).toBe(1);
  });
  it('should render TaskItem', () => {
    const taskColumnT = shallow(
      <TasksColumn
        sortedTasks={[{
          id: '-Lcf8EcnE8OeEYXCWiKT',
          name: 'TEST2',
        }]}
        title="To Do"
      />,
    );
    const tasksCont = taskColumnT.find('[data-test="taskItemsContainer"]');
    expect(tasksCont.children().length).toBe(1);
    expect(tasksCont.children().name()).toBe('TaskItem');
  });
});
