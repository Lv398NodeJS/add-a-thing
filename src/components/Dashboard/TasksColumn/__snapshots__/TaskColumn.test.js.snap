// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskColumn /> should render and match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "tasks-column rounded mb-4 mb-lg-0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "tasks-column-invalid-title",
          "data-test": "columnTitle",
        },
        "children": Array [],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "task-items-container container-fluid",
            "data-test": "taskItemsContainer",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "task-items-container container-fluid",
          "data-test": "taskItemsContainer",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "tasks-column-invalid-title",
            "data-test": "columnTitle",
          },
          "children": Array [],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
